2016/exercises/final_project/Alba-Petar-Rebecca-Simon/PolytopeProject/Gale_Diagram.cpp

```diff
@@ -0,0 +1,322 @@
+//
+//  Gale_Diagram.cpp
+//  PolytopeProject
+//
+
+
+#include "Gale_Diagram.hpp"
+
+
+
+Gale_Diagram::Gale_Diagram(vector<Point> points): points(points),matrix(Matrix(points)){
+};
+
+bool Gale_Diagram::is_neighborly() const{
+    //Choose 2 points
+    for(int i=0; i<points.size()-1; i++){
+        for(int j=i+1; j<points.size(); j++){
+            
+            bool edgeIJPresent=false;
+            
+            
+            //Then choose another 2
+            for(int k=0; k<points.size()-1 && (!edgeIJPresent); k++){
+                for(int l=k+1; l<points.size() && (!edgeIJPresent); l++){
+					if( (i!=k) and (i!=l) and (j!=k) and (j!=l)){
+						edgeIJPresent = isSimplex(i, j, k, l);
+					}
+                }
+            }
+            
+            if(!edgeIJPresent){
+                return false;
+            }
+        }
+        
+    }
+    return true;
+}
+
+
+
+vector<int> Gale_Diagram::findMissing(int i, int j, int k, int l) const{
+    vector<int> pointNums(points.size());
+    for(int a=0; a<points.size(); a++) pointNums[a]=a;
+
+	pointNums[i] = -1;
+	pointNums[j] = -1;
+	pointNums[k] = -1;
+	pointNums[l] = -1;
+	
+    vector<int> pN;
+    
+    for(int a=0; a<points.size(); a++){
+		if(pointNums[a]>=0) pN.push_back(a);
+	}
+    
+    return pN;
+}
+
```

Comment:
You're function uses the methods resize or distance, which are easy readable, but which I didn't know. I didn't know copy_if either, but I had to look at the documentation to know what it does. Also I don't know what [&] means (more specifically why you use straight brackets).

For the name, if you consider the numbers from 0 to 7 and have 4 of them (i,j,k,l) then find the other (missing) four. Considering the gale diagram then i,j,k,l indeed are the vertices that will be removed. So the naming could indeed be improved.